MagicBytesValidator/Services/Http/IFormFileTypeProvider.cs

```diff
@@ -14,11 +21,35 @@ public interface IFormFileTypeProvider
     /// <summary>
     /// Tries to find matching FileType for given IFormFile.
     /// </summary>
-    /// <param name="formFile">Given IFormFile</param>
-    /// <returns>Matching FileType (if known)</returns>
     /// <exception cref="MimeTypeMismatchException">
     /// When file-type by extension and given content-type (IFormFile.ContentType) differ.
     /// In this case, someone <i>could</i> try to circumvent the validation.
     /// </exception>
-    FileType? FindFileTypeForFormFile(IFormFile formFile);
+    [Obsolete("Use FindValidatedType instead.")]
+    IFileType? FindFileTypeForFormFile(IFormFile formFile);
+
+    /// <summary>
+    /// Tries to find matching <see cref="IFileType"/> for given <see cref="IFormFile"/> that also matches
+    /// the content of the form file.
+    /// </summary>
+    /// <param name="formFile"><see cref="IFormFile"/> that the <see cref="IFileType"/> should be found for</param>
+    /// <param name="formFileStream">
+    /// Optional. If the file stream for the form file is already loaded, it can be included here.
+    /// This prevents opening a read stream for the same file multiple times.
+    /// <b>However, never include streams of other files than the given form file! Otherwise the validation may be
+    /// wrong and could be circumvented!</b>
+    /// </param>
+    /// <param name="cancellationToken">CancellationToken</param>
+    /// <returns>
+    /// <see cref="IFileType"/> that matches by the form files content type, content (and extension, if given)
```

Comment:
```suggestion
    /// <see cref="IFileType"/>That matches by the form files content type, content (and extension, if given)
```

Reply:
The comment actually looks correct ("Returns `IFileType` that ma...") when visible in the IDE